### DISCUSSION TITLE
Support more filenames besides Jenkinsfile and *.groovy

### DISCUSSION THREAD

**Discussion by zizzencs** [lppedd/idea-jenkins-pipeline#3] [General]
In `jenkinsPipeline.gdsl` the root context is defined to be in a `Jenkinsfile` or in `*.groovy`. I created a small modification to match on `Jenkinsfile*` instead, as we use `Jenkinsfile.build`, `Jenkinsfile.deploy` and similar files as Jenkinsfiles. I'd suggest implementing something similar for this plugin, or at least making it possible to mark other files as Jenkinsfiles.

In `JPUtils.java` I did a similar thing, changed the Jenkinsfile detection mechanism to `virtualFile.getName().startsWith(JPConstants.Files.JENKINSFILE)` to have nice icons for our Jenkinsfiles. I'd also suggest implementing something similar.

Keep up with the good work, this plugin is already extremely useful.

**Comment by lppedd** [comment]
Hi @zmezei-dpc! Sorry for not providing a ready-to-use binary, still working on it and had a bit of stuff to do for my daily job.

Yes that is a good idea. But I think the beauty of dealing with GDSL internally is we can do pretty much what we want. In this specific case I can simply have a placeholder and let the user specify the filename pattern, then load the GDSL (or reload it using the appropriate service).

Ideally, Jenkinsfile would be a `FileType` on its own, so that we could re-use the custom file association settings.
